nexus/db-queries/src/db/datastore/bgp.rs

```diff
@@ -285,11 +287,12 @@ impl DataStore {
                             }),
                     }?;
 
-                    let count =
+                    let count: i64 =
                         sps_bgp_peer_config_dsl::switch_port_settings_bgp_peer_config
                         .filter(sps_bgp_peer_config::bgp_config_id.eq(id))
+                        .select(sps_bgp_peer_config::bgp_config_id)
                         .count()
-                        .execute_async(&conn)
+                        .get_result_async(&conn)
```

Comment:
You don't need to select all the rows if you just want to check whether there are any. Something like this should do it:

```rs
let configs_exist = diesel::dsl::select(diesel::dsl::exists(
  sps_bgp_peer_config_dsl::switch_port_settings_bgp_peer_config
    .filter(sps_bgp_peer_config::bgp_config_id.eq(id))
))
.get_result_async::<bool>(&conn)
.await?;
```

Compare:

https://github.com/oxidecomputer/omicron/blob/d00b684ff16d986c80a9e603fa7c5b53a8695144/nexus/db-queries/src/db/datastore/ip_pool.rs?plain=1#L892-L898

Reply:
Done